Khan/gomacro
# Add limited ability to import local modules

# Summary
This change adds the ability to import local modules without having to fetch them IFF the module being imported uses a `go.mod` file that is in the directory `gomacro` was run from, or is located in one of its parent directories. 

It facilitates this by modifying the package `go.mod` file used by `gomacro` for said package to include a replace directive pointing to the local location. It also copies the replace directives of the source modules `go.mod` file, so replacements in that source module are used by the imported package as well.

This change is somewhat limited in its abilities, but sets us up to expand how we determine if an imported module can and/or should be sourced from local disk or remote repository. We should be able to support `./` and `../` style imports with some minor modifications to `genimport.local` - though there is a check elsewhere in the code that actually prevents using such imports which I haven't yet tracked down. We could alternately introduce a `:replace` interpreter directive that could be used to modify this as well, giving us finer and more go.mod-like control over how we perform imports.

# Testing
1. Make a change to a locally available module
2. From that modules root, run `gomacro`, then import the package in the REPL via `import "github.com/user/project/path/to/pkg"`. Observe the change is available.
3. Repeat the above step from a subdirectory, not necessarily on the same path as the desired pkg. Observe the local code is still used.
4. Repeat the above step from another directory that is not a child of the project, and observe that the package still loads, but the publicly available package is used instead.
